### PR TITLE
Fix ignore hidden files when generating datasets

### PR DESCRIPTION
**Pull request by kashif**
the data sets contain ._ json mac osx specific hidden files which get read when regenrating datasets e.g. "taxi_30min"

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/dataset/repository/_gp_copula_2019.py [hunk 1]
```diff
@@ -161,4 +161,4 @@ def get_data(dataset_path: Path, ds_info: GPCopulaDataset):
 
 def clean_up_dataset(dataset_path: Path, ds_info: GPCopulaDataset):
     os.remove(dataset_path.parent / f"{ds_info.name}.tar.gz")
-    shutil.rmtree(dataset_path / "metadata")
+    shutil.rmtree(dataset_path / "metadata", ignore_errors=True)
```
Comment (lostella) [inline src/gluonts/dataset/repository/_gp_copula_2019.py:164]: Is this also needed?

Comment (kashif) [reply src/gluonts/dataset/repository/_gp_copula_2019.py:164]: yes was getting an error with get_dataset("taxi_30min", regenerate=True) without it... somehow the metadata/ folder gets removed before... I didn't figure it out...

File: src/gluonts/dataset/common.py [hunk 2]
```diff
@@ -156,7 +156,7 @@ def FileDataset(
     use_timestamp: bool = False,
     loader_class=None,
     ignore=False,
-    pattern="*",
+    pattern="[!._]*",
```
Comment (lostella) [inline src/gluonts/dataset/common.py]: Instead of making this the global default, we could set this option in line 157 of `_gp_copula_2019.py` maybe?

**Review by lostella — CHANGES_REQUESTED** [review]

File: src/gluonts/dataset/common.py (hunk 2 above)
Comment (kashif) [reply src/gluonts/dataset/common.py]: ok yes sure

**Review by lostella — APPROVED** [review]
Thanks @kashif!

**Pull request merged**